src/tools/apollo-tool/apollo-tuner.cpp

```diff
@@ -231,37 +253,43 @@ extern "C" void kokkosp_request_values(
   variableSet tuningProblem;
   int numValidVariables = 0;
   for (int x = 0; x < numContextVariables; ++x) {
-    if (untunables.find(contextValues[x].id) == untunables.end()) {
-      tuningProblem.variable_ids[numValidVariables++] = contextValues[x].id;
+    if (untunables.find(contextValues[x].type_id) == untunables.end()) {
+      tuningProblem.variable_ids[numValidVariables++] = contextValues[x];
     }
   }
   for (int x = 0; x < numTuningVariables; ++x) {
-    if (unlearnables.find(tuningValues[x].id) == unlearnables.end()) {
-      tuningProblem.variable_ids[numValidVariables++] = tuningValues[x].id;
+    if (unlearnables.find(tuningValues[x].type_id) == unlearnables.end()) {
+      tuningProblem.variable_ids[numValidVariables++] = tuningValues[x];
     }
   }
   tuningProblem.num_variables = numValidVariables;
   if (tuning_regions.find(tuningProblem) == tuning_regions.end()) {
     std::cout << "Testing space size is " << choiceSpaceSize << std::endl;
-    std::string name = "debug_name_"+std::to_string(created_regions);
-    auto region =
-        new Apollo::Region(numContextVariables, name.c_str(), choiceSpaceSize);
-    tuning_regions[tuningProblem] = region;
+    std::string name = get_region_name(tuningProblem);
+    Apollo::Region* region;
+    if(file_exists(name)) {
+      region = new Apollo::Region(numContextVariables,name.c_str(), choiceSpaceSize, name); 
+    }
+    else{
+      region =
+          new Apollo::Region(numContextVariables, name.c_str(), choiceSpaceSize);
+    }
+    tuning_regions[tuningProblem] = std::make_pair(name,region);
   }
-  auto region = tuning_regions[tuningProblem];
+  auto region = tuning_regions[tuningProblem].second;
   tuned_contexts[contextId] = region;
   region->begin();
   //std::cout <<"Cont: "<<numContextVariables<<std::endl;
   for (int x = 0; x < numContextVariables; ++x) {
-    if (untunables.find(contextValues[x].id) == untunables.end()) {
+    if (untunables.find(contextValues[x].type_id) == untunables.end()) {
       region->setFeature(variableToFloat(contextValues[x]));
     }
   }
 
   int policyChoice = region->getPolicyIndex();
```

Comment:
See above. I think int is fine here... unless apollo returns a larger integer.  I would probably `auto` this just in case Apollo adds support for even larger search spaces and we want to handle >1e9 search spaces.

Reply:
I like that. Future-proof us if Apollo has that ability. I'm not sure OpenCV (which Apollo uses) allows that to be an int64, but if they do, that's a win